### DISCUSSION TITLE
Export feature for all config (compose.yaml and .env)

### DISCUSSION THREAD

**Discussion by matbgn** [louislam/dockge#308] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

A tool that could open backup capability to avoid multiple users error

### ✔️ Solution

A single flat file or maybe a zip file for all compose. Any idea will be welcomed

### ❓ Alternatives

At the very first, only download option could already be super useful to facilitate git review when someone is changing something on Dockge UI

**Comment by trythatagain** [comment]
You could set a cron job to backup the compose directory at regular intervals, or setup Syncthing which would sync when any changes are made. Or am I not getting what you want to do?

**Reply by matbgn** [reply]
Not really, if you're multiple users managing the same Dockge instance, having the export capability gives you the possibility to update/(possibly in the future rollback) the changes because you can track it for example in a git PR or whatever solution you want to make the diff in an easy way.

Many docker management tools like CapRover, Portainer, etc. give you the ability to back up configs, and considering that Dockge already has access to data in flat mode, I expected that at least for the download part this would be absolutely no problem. Isn't it? If not, and in a constructive way, what are the steps to be fulfil that a pull request could be accepted in Dockge?

**Comment by handy-git** [comment]
Very good suggestion. Please provide import and export for configurations.

**Comment by badabing2005** [comment]
I had one of my servers crash, which had lots of containers.
Many of those compose files were hours / days of fine tuning.
I'm guilty of not backing up elsewhere.
I had setup dockge on it, which was connected to main dockge instance on another server.

When I open the console in dockge UI, I can see those compose files in the docker directory, but the console does not allow me to cat or tar or any way to export or expose them.

Is there a manual way I can extract those? it would save me days, and even then, I might not get it the same.
Thanks

**Comment by trythatagain** [comment]
Can you not do as I suggest further up this thread, either syncthing or a simple script with a cron job.
Or is it that you can't ssh in and access the compose files?

**Comment by badabing2005** [comment]
Perhaps I'm not understanding properly or not explaining correctly.
The server that crashed is inaccessible, hence I cannot connect to dockge conatiner on that server.
But the server was part of dockge cluster (if you may call it that way)
What I'm trying to extract is cached copies of the compose files on another dockge, perhaps they're in db or elsewhere.
Or perhaps there is no such thing as cached copies.

Thanks

**Comment by trythatagain** [comment]
Ah right, I believe all the compose.yml files are in the folder you setup for dockage (DOCKGE_STACKS_DIR=/compose for example), so if your compose folder was on that inaccessible drive, then I'm afraid they're lost.
Can you remove the drive and connect it via usb sata and see if you can access the folder from linux?

**Comment by badabing2005** [comment]
Thanks for the follow up.
The drive is totally inaccessible, doesn't even get detected by recovery software :(
If there is no cache in dockge then they're gone for good. :(
